File: src/structural/bridge2.java
public abstract class Pizza {
    protected String sauce;
    protected String toppings;
    protected String crust;

    public abstract void deliver();
}

public class ItalianVeggiePizza extends Pizza {
    @Override
    public void deliver() {
        System.out.println("Adding Oil Sauce");
        System.out.println("Adding No Toppings");
        System.out.println("Adding Cheese");
        System.out.println("Crust is: Thin");
        System.out.println("Order in Progress!");
    }
}

public class ItalianPepperoniPizza extends Pizza {
    @Override
    public void deliver() {
        System.out.println("Adding Oil Sauce");
        System.out.println("Adding No Toppings");
        System.out.println("Adding Pepperoni");
        System.out.println("Crust is: Thin");
        System.out.println("Order in Progress!");
    }
}

public class AmericanVeggiePizza extends Pizza {
    @Override
    public void deliver() {
        System.out.println("Adding Super Secret Sauce");
        System.out.println("Adding Toppings: Everything");
        System.out.println("Adding Cheese");
        System.out.println("Crust is: Thick");
        System.out.println("Order in Progress!");
    }
}

public class AmericanPepperoniPizza extends Pizza {
    @Override
    public void deliver() {
        System.out.println("Adding Secret Sauce");
        System.out.println("Adding All Toppings");
        System.out.println("Adding Pepperoni");
        System.out.println("Crust is: Thick");
        System.out.println("Order in Progress!");
    }
}

File: src/structural/bridge2.ts
abstract class Pizza {
  constructor(
    protected sauce: string,
    protected toppings: string,
    protected crust: string,
  ) {}

  abstract deliver(): void;
}

export class ItalianVeggiePizza extends Pizza {
  deliver(): void {
    console.log("Adding Oil Sauce");
    console.log("Adding No Toppings");
    console.log("Adding Cheese");
    console.log("Crust is: Thin");
    console.log("Order in Progress!");
  }
}

export class ItalianPepperoniPizza extends Pizza {
  deliver(): void {
    console.log("Adding Oil Sauce");
    console.log("Adding No Toppings");
    console.log("Adding Pepperoni");
    console.log("Crust is: Thin");
    console.log("Order in Progress!");
  }
}

export class AmericanVeggiePizza extends Pizza {
  deliver(): void {
    console.log("Adding Super Secret Sauce");
    console.log("Adding Toppings: Everything");
    console.log("Adding Cheese");
    console.log("Crust is: Thick");
    console.log("Order in Progress!");
  }
}

export class AmericanPepperoniPizza extends Pizza {
  deliver(): void {
    console.log("Adding Secret Sauce");
    console.log("Adding All Toppings");
    console.log("Adding Pepperoni");
    console.log("Crust is: Thick");
    console.log("Order in Progress!");
  }
}
